modules/mbedtls/Kconfig.psa

```diff
@@ -179,11 +159,13 @@ config PSA_HAS_HASH_SUPPORT
 		   PSA_WANT_ALG_SHA_512		|| \
 		   PSA_WANT_ALG_RIPEMD160	|| \
 		   PSA_WANT_ALG_MD5
+	help
+	  Prompt-less configuration that states that hash is supported.
 
 config PSA_WANT_ALG_SHA_1
 	bool
 	prompt "PSA SHA1 support" if !PSA_PROMPTLESS
-	default y if !PSA_DEFAULT_OFF
+	default y if !PSA_DEFAULT_OFF && !BUILD_WITH_TFM
```

Comment:
"The default for PSA_WANT_ALG_SHA_1 is adjusted to
 not enable this when BUILD_WITH_TFM is enabled. Note that the
 defconfig in nrf (kconfig.psa.defconfig) for TF-M is already set
 to not enable SHA-1 by default, but this isn't enough to turn the
 feature off."

Why is this not enough? I would have thought it would be enough.

Reply:
Because the crypto-backends don't look at these configurations. They just call `hash_compute`.
So we need an explicit "NOT_SUPPORTED" return before calling those functions.

```
#if !defined(PSA_WANT_ALG_SHA_1)
    if (alg == PSA_ALG_SHA_1) {
        return PSA_ERROR_NOT_SUPPORTED;
    }
#endif
...
#if defined(PSA_CRYPTO_DRIVER_HAS_HASH_SUPPORT_CC3XX)
    status = cc3xx_hash_compute(alg, input, input_length, hash, hash_size,
            hash_length);
...
#endif /* PSA_CRYPTO_DRIVER_HAS_HASH_SUPPORT_CC3XX */
#if defined(PSA_CRYPTO_DRIVER_HAS_HASH_SUPPORT_OBERON)
    status = oberon_hash_compute(alg, input, input_length, hash, hash_size,
            hash_length);
...
#endif /* PSA_CRYPTO_DRIVER_HAS_HASH_SUPPORT_OBERON */
```

Reply:
Should we delete Kconfig.psa.defconfig then?

Since it is not executed early enough to have any effect?

Or is there some configuration where it does have effect?